content/en/docs/components/pipelines/sdk/component-development.md

```diff
@@ -92,296 +100,459 @@ def do_work(input1_file, output1_file, param1):
+
+# Compile, upload, and submit this pipeline for execution.
+client.create_run_from_pipeline_func(my_pipeline, arguments={})
 ```
 
+[kfp-load-comp-file]: https://kubeflow-pipelines.readthedocs.io/en/stable/source/kfp.components.html#kfp.components.load_component_from_file
+[kfp-load-comp-url]: https://kubeflow-pipelines.readthedocs.io/en/stable/source/kfp.components.html#kfp.components.load_component_from_url
+[kfp-load-comp-text]: https://kubeflow-pipelines.readthedocs.io/en/stable/source/kfp.components.html#kfp.components.load_component_from_text
+[kfp-containerop]: https://kubeflow-pipelines.readthedocs.io/en/stable/source/kfp.dsl.html#kfp.dsl.ContainerOp
+
 ## Organizing the component files
 
-This section provides a recommended way to organize the component files. There
-is no requirement that you must organize the files in this way. However, using
+This section provides a recommended way to organize a component's files. There
+is no requirement that you must organize the files in this way. However, using 
 the standard organization makes it possible to reuse the same scripts for
-testing, image building and component versioning.
-See this
-[sample component](https://github.com/kubeflow/pipelines/tree/master/components/sample/keras/train_classifier)
-for a real-life component example.
+testing, image building, and component versioning.
 
 ```
 components/<component group>/<component name>/
 
-    src/*            #Component source code files
-    tests/*          #Unit tests
-    run_tests.sh     #Small script that runs the tests
-    README.md        #Documentation. Move to docs/ if multiple files needed
+    src/*            # Component source code files
+    tests/*          # Unit tests
+    run_tests.sh     # Small script that runs the tests
+    README.md        # Documentation. If multiple files are needed, move to docs/.
 
-    Dockerfile       #Dockerfile to build the component container image
-    build_image.sh   #Small script that runs docker build and docker push
+    Dockerfile       # Dockerfile to build the component container image
+    build_image.sh   # Small script that runs docker build and docker push
 
-    component.yaml   #Component definition in YAML format
+    component.yaml   # Component definition in YAML format
 ```
 
+See this [sample component][org-sample] for a real-life component example.
+
+[org-sample]: https://github.com/kubeflow/pipelines/tree/master/components/sample/keras/train_classifier
+
 ## Next steps
 
 * Consolidate what you've learned by reading the
   [best practices](/docs/components/pipelines/sdk/best-practices) for designing and
   writing components.
 * For quick iteration,
-  [build lightweight components](/docs/components/pipelines/sdk/lightweight-python-components/)
+  [build lightweight Python function-based components](/docs/components/pipelines/sdk/python-function-components/)
   directly from Python functions.
 * See how to [export metrics from your
```

Comment:
Not part of the scope of your update, but this link for exporting metrics gives me a "page not found" message.

Reply:
Thanks for catching this!